src/agent/advertising_proxy.cpp

```diff
@@ -230,6 +242,7 @@ void AdvertisingProxy::PublishHostHandler(const char *aName, otbrError aError)
         }
         else
         {
+            update->mHostName.clear();
```

Comment:
```suggestion
            update->mHostNamePublished = true;
```

Maybe add a flag rather than clear the host name?